controller/engine_image_controller.go

```diff
@@ -536,6 +554,17 @@ func (ic *EngineImageController) enqueueAllEngineImagesForNodeChange() {
 	return
 }
 
+func (ic *EngineImageController) enqueueInstanceManager(im *longhorn.InstanceManager) {
+	ei, err := ic.ds.GetEngineImage(im.Spec.EngineImage)
```

Comment:
Do we need this?
```
if im.Spec.OwnerID != ic.controllerID {
    return
}
```

Reply:
https://github.com/longhorn/longhorn-manager/commit/b26e730f12145cf55ef1731703b5104023d25638

Reply:
Only the owner of the engine image updates the engine image status. We don't really know who is the owner here? 

IIRC, instance manager's owner has no relationship with engine image's owner. I am not sure we need this line here.

Reply:
Engine image controller will handle all instance manager regardless of the NodeID:
https://github.com/longhorn/longhorn-manager/blob/e5ecc6c96a1abf519e7324721827f4036115a444/controller/engine_image_controller.go#L636

But instance manager will be updated frequently, and most of the updates (involving the instance map update) are meaningless to the engine image itself. I am afraid of this informer will increase CPU usage.

Reply:
Yes. There is no relationship between the engine image's owner and instance manager's owner.

Reply:
Shouldn't be too bad. Unless there is an easy way to avoid the informer, we probably have to do it, at least for now.

Reply:
OK. Then we don't need to check `OwnerID`